core/src/test/java/org/apache/oozie/command/wf/TestSignalXCommand.java

```diff
@@ -314,4 +322,98 @@ public boolean evaluate() throws Exception {
         assertEquals(prepActions.length, numPrep);
         assertEquals(okActions.length, numOK);
     }
+
+    private void writeToFile(String appXml, String appPath) throws IOException {
+        File wf = new File(URI.create(appPath));
+        PrintWriter out = null;
+        try {
+            out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(wf), StandardCharsets.UTF_8));
+            out.println(appXml);
+        }
+        catch (IOException iex) {
+            throw iex;
+        }
+        finally {
+            if (out != null) {
+                out.close();
+            }
+        }
```

Comment:
You could simply use:

        try(PrintWriter out = new PrintWriter(new OutputStreamWriter(new FileOutputStream(wf), StandardCharsets.UTF_8))) {
            out.println(appXml);
        }